src/lang/std/engineConnection.ts

```diff
@@ -1667,6 +1689,27 @@ export class EngineCommandManager {
       promise,
       resolve,
     }
+    if (command.type === 'extrude') {
+      this.artifactMap[id] = {
+        range: range || [0, 0],
+        pathToNode,
+        type: 'pending',
+        commandType: 'extrude',
+        parentId: getParentId(),
+        promise,
+        target: command.target,
+        resolve,
+      }
+      const target = this.artifactMap[command.target]
+      if (target.commandType === 'start_path') {
+        const temp = target as any
+        if (temp?.extrusions?.length) {
+          temp.extrusions.push(id)
+        } else {
+          temp.extrusions = [id]
+        }
```

Comment:
I'm happy to merge as it, but wondering if the typing I was attempting with `CommandInfo & {...` is a bad way of doing the types?

Reply:
I'll take a look at this today, yeah I'd like to see if I can figure out a typing for it now that I've got some sleep

Reply:
Eh I made the type coercion we do a bit more explicit, but I was struggling to find a way to update the approach so that `CommandInfo` doesn't need to be an intersection type (like trying a generic `dependents` array for example). I think it's definitely something to come back around to because I believe the problem of tracking down the dependencies for any operation is going to be generically important.